src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationInvalidEnvironmentException.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+using System;
+
+namespace AspNet.Security.OAuth.SuperOffice
+{
+    /// <summary>
+    /// The exception that is thrown when the <see cref="SuperOfficeAuthenticationOptions"/> Environment property is not valid.
+    /// </summary>
+    public class SuperOfficeAuthenticationInvalidEnvironmentException : Exception
```

Comment:
I don't think we need to add a whole custom exception type for this one use case. `InvalidOperationException` or `NotSupportedException` would suffice.